packages/webpack/mixin.core.js

```diff
@@ -63,11 +63,40 @@ class WebpackMixin extends Mixin {
     const { StatsFilePlugin } = require('./lib/plugins/stats');
     return new StatsFilePlugin(this);
   }
-  getBuildConfig(target) {
+  _oldGetBuildConfig(target) {
+    console.warn('getBuildConfig with target:', target, 'is deprecated.');
```

Comment:
We should consider using https://www.npmjs.com/package/depd for this and have a `deprecations.md` file in the repository root where we describe the deprecations in detail: https://github.com/nodejs/node/blob/master/doc/api/deprecations.md